debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/util/TestHelper.java

```diff
@@ -345,4 +350,66 @@ private static ObjectName getObjectName(String context, String serverName) throw
     public static int waitTimeForRecords() {
         return Integer.parseInt(System.getProperty(TEST_PROPERTY_PREFIX + "records.waittime", "5"));
     }
+
+    /**
+     * Utility method that will poll the CDC change tables and provide the record handler with the changes detected.
+     * The record handler can then make a determination as to whether to return {@code true} if the expected outcome
+     * exists or {@code false} to indicate it did not find what it expected.  This method will block until either
+     * the handler returns {@code true} or if the polling fails to complete within the allocated poll window.
+     *
+     * @param connection the SQL Server connection to be used
+     * @param tableName the main table name to be checked
+     * @param handler the handler method to be called if changes are found in the capture table instance
+     */
+    public static void waitForCdcRecord(SqlServerConnection connection, String tableName, CdcRecordHandler handler) {
+        try {
+            Awaitility.await("Checking for expected record in CDC table for " + tableName)
+                    .atMost(30, TimeUnit.SECONDS)
+                    .pollDelay(Duration.ofSeconds(0))
+                    .pollInterval(Duration.ofSeconds(1)).until(() -> {
+                        if (!connection.getMaxLsn().isAvailable()) {
+                            return false;
+                        }
+
+                        for (SqlServerChangeTable ct : connection.listOfChangeTables()) {
+                            final String ctTableName = ct.getChangeTableId().table();
+                            if (ctTableName.endsWith("dbo_" + connection.getNameOfChangeTable(tableName))) {
+                                try {
+                                    final Lsn minLsn = connection.getMinLsn(ctTableName);
+                                    final Lsn maxLsn = connection.getMaxLsn();
+                                    final AtomicReference<Boolean> found = new AtomicReference(false);
+                                    SqlServerChangeTable[] tables = Collections.singletonList(ct).toArray(new SqlServerChangeTable[]{});
+                                    connection.getChangesForTables(tables, minLsn, maxLsn, resultsets -> {
+                                        final ResultSet rs = resultsets[0];
+                                        while (rs.next()) {
+                                            if (handler.apply(rs)) {
+                                                found.set(true);
```

Comment:
Extracting a method might help to avoid having to modify the atomic reference as a side-effect. We do that elsewhere too and I generally don't really like it.